app/src/main/java/org/thoughtcrime/securesms/components/emoji/EmojiProvider.java

```diff
@@ -36,6 +36,25 @@ public class EmojiProvider {
   private static final    String TAG   = Log.tag(EmojiProvider.class);
   private static final    Paint  PAINT = new Paint(Paint.FILTER_BITMAP_FLAG | Paint.ANTI_ALIAS_FLAG);
 
+  public interface onCandidatesLoadedListener {
+    void onCandidatesLoaded(EmojiParser.CandidateList candidates);
+  }
+
+  public static void getCandidates(@Nullable CharSequence text, onCandidatesLoadedListener onCandidatesLoaded) {
```

Comment:
IMO `getCandidates` is a somewhat misleading name as I expect it to return something. Maybe `loadCandidates` to more closely match the `onCandidatesLoaded` method in the interface?